src/account/sync/mod.rs

```diff
@@ -863,22 +863,47 @@ impl SyncedAccount {
                 balance: a.available_balance(&account),
             })
             .collect();
-        let addresses = input_selection::select_input(threshold, available_addresses)?;
+        let mut selected_addresses = input_selection::select_input(transfer_obj.amount.get(), available_addresses)?;
+        let has_remainder = selected_addresses.iter().fold(0, |acc, a| acc + a.balance) > transfer_obj.amount.get();
+
+        // if we're reusing the input address for remainder output
+        // and we have remainder value, we should run the input selection again
+        // without the output address.
+        if has_remainder
+            && transfer_obj.remainder_value_strategy == RemainderValueStrategy::ReuseAddress
+            && addresses.iter().any(|input| input.address() == &transfer_obj.address)
+        {
+            let available_addresses: Vec<input_selection::Input> = addresses
+                .iter()
+                .filter(|a| {
+                    // we do not allow the deposit address as input address
+                    a.address() != address
```

Comment:
Will the consolidation still work with this?

Reply:
yeah this is only a fallback